### PR TITLE
EIP-2274: Multiple address spaces for Precompiled contracts

### PR DESCRIPTION
**Pull request by AntoineRondelet**

### CODE REVIEW

File: EIPS/eip-2274.md [hunk 1]
```diff
@@ -6,23 +6,24 @@ discussions-to: https://ethereum-magicians.org/t/eip-2274-multiple-addresses-ran
 status: Draft
 type: Standards Track
 category: Core
-requires: EIP-1109
+requires: 1109
+replaces: 1352
```
Comment (axic) [inline EIPS/eip-2274.md]: 1352 has a very different purpose so I can't see how this replaces it.

Comment (AntoineRondelet) [reply EIPS/eip-2274.md]: That's right @axic my bad. I'll update the document accordingly. Thanks

File: EIPS/eip-2274.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Other EIPs can choose to include links to test cases if applicable.-->
+Same as in [eip-1109](https://eips.ethereum.org/EIPS/eip-1109), with:
+- Call to defined and undefined precompiled for different `vmID` values.
+
+## Implementation
+<!--The implementations must be completed before any EIP is given status "Final", but it need not be completed before the EIP is accepted. While there is merit to the approach of reaching consensus on the specification and rationale before writing code, the principle of "rough consensus and running code" is still useful when it comes to resolving many discussions of API details.-->
+In order to support this new OPCODE and **illustrate** the point made above, we propose an **incomplete and broken** implementation below:
+- In `core/vm/instructions.go` (see: [here](https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go)), add a function for the new OPCODE:
```
Comment (shemnon) [inline EIPS/eip-2274.md]: Putting geth code in an EIP presents licensing problems as it is LGPL licensed and has a copyleft clause which compels derivative works to be placed under LGPL.  That copyleft clause is incompatible with the EIP requirement that all EIPs be public domain works.

Consider either switching to pseudo code or creating a PR and referencing the PR in this EIP.

Comment (AntoineRondelet) [reply EIPS/eip-2274.md]: Thanks @shemnon, I removed the code and added links to a fork of geth containing the functions.

**Pull request closed without merging**